keras/backend.py

```diff
@@ -5574,6 +5574,88 @@ def categorical_crossentropy(target, output, from_logits=False, axis=-1):
     return -tf.reduce_sum(target * tf.math.log(output), axis)
 
 
+@keras_export("keras.backend.categorical_focal_crossentropy")
+@tf.__internal__.dispatch.add_dispatch_support
+@doc_controls.do_not_generate_docs
+def categorical_focal_crossentropy(
+    target,
+    output,
+    alpha=0.25,
+    gamma=2.0,
+    from_logits=False,
+    axis=-1,
+):
+    """Computes the alpha balanced focal crossentropy loss.
+
+    According to [Lin et al., 2018](https://arxiv.org/pdf/1708.02002.pdf), it
+    helps to apply a focal factor to down-weight easy examples and focus more on
+    hard examples. By default, the focal tensor is computed as follows:
+
+    It has pt defined as:
+    pt = p, if y = 1 else 1 - p
+
+    The authors use alpha-balanced variant of focal loss in the paper:
+    FL(pt) = −α_t * (1 − pt)^gamma * log(pt)
+
+    Extending this to multi-class case is straightforward:
+    FL(pt) = α_t * (1 − pt)^gamma * CE, where minus comes from
+    negative log-likelihood and included in CE.
+
+    `modulating_factor` is (1 − pt)^gamma, where `gamma` is a focusing
+    parameter. When `gamma` = 0, there is no focal effect on the categorical
+    crossentropy. And if alpha = 1, at the same time the loss is equivalent
+    to the categorical crossentropy.
+
+    Args:
+        target: A tensor with the same shape as `output`.
+        output: A tensor.
+        alpha: A weight balancing factor for all classes, default is `0.25` as
+            mentioned in the reference. It can be a list of floats or a scalar.
+            In the multi-class case, alpha may be set by inverse class
+            frequency by using `compute_class_weight` from `sklearn.utils`.
+        gamma: A focusing parameter, default is `2.0` as mentioned in the
+            reference. It helps to gradually reduce the importance given to
+            simple examples in a smooth manner.
+        from_logits: Whether `output` is expected to be a logits tensor. By
+            default, we consider that `output` encodes a probability
+            distribution.
+
+    Returns:
+        A tensor.
+    """
+    target = tf.convert_to_tensor(target)
+    output = tf.convert_to_tensor(output)
+    target.shape.assert_is_compatible_with(output.shape)
+
+    output, from_logits = _get_logits(
+        output, from_logits, "Softmax", "categorical_focal_crossentropy"
+    )
+
+    output = tf.__internal__.smart_cond.smart_cond(
+        from_logits,
+        lambda: softmax(output),
+        lambda: output,
+    )
+
+    # scale preds so that the class probas of each sample sum to 1
```

Comment:
Again this was taken from `categorical_crossentropy`:

https://github.com/keras-team/keras/blob/c72e310ab5452e5a8bbcc2f18ab092f21d8846b2/keras/backend.py#L5569-L5570

I can change this to: "Adjust the predictions so that the probability of each class for every sample adds up to 1."